repl_tools/src/lib.rs

```diff
@@ -0,0 +1,250 @@
+use std::{
+    borrow::Cow,
+    cell::Cell,
+    marker::PhantomData,
+    path::{Path, PathBuf},
+    rc::Rc,
+};
+
+use rustyline::{
+    completion::{Completer, FilenameCompleter, Pair},
+    highlight::Highlighter,
+    hint::Hinter,
+    validate::Validator,
+    Helper,
+};
+
+#[doc(hidden)]
+pub use rustyline as __rustyline;
+
+pub trait HighlightAndComplete: Sized {
+    fn from_str(line: &str) -> Result<Self, Box<dyn std::error::Error>>;
```

Comment:
I think this might be a good place to start introducing concrete error types. I'd suggest we try using [`anyhow`](https://docs.rs/anyhow/), but no preference here (and this can be done at a later time, anyway).

Reply:
`anyhow` is for programs, not libraries. It doesn't really have a way to match on the contained error.